src/ir.cpp

```diff
@@ -12095,7 +12098,10 @@ static bool ir_analyze_fn_call_generic_arg(IrAnalyze *ira, AstNode *fn_proto_nod
             return false;
         }
 
-        casted_args[fn_type_id->param_count] = casted_arg;
+        if (*casted_args == nullptr) {
+            *casted_args = allocate<IrInstruction *>(1);
```

Comment:
If we have not yet allocated memory for arguments (because upfront there were none to allocate), allocate memory now.